### PR TITLE
[FLOC 3568] Remove the MongoDB tutorial and references to the Vagrant Tutorial Box

### PR DESCRIPTION
**Pull request by davidoliverSP2**
Fixes 3568

Due to the Vagrant tutorial box being broken, the existing MongoDB tutorial, and associated topics (Flocker Examples) have been removed. 

In addition, all references to the Vagrant Tutorial box have also been removed.

These are due to be replaced with Flocker Integration Tutorials (currently in planning phase, and not yet documented as Jira issues/epics/stories)

### CODE REVIEW

File: docs/gettinginvolved/infrastructure/vagrant.rst [hunk 1]
```diff
@@ -9,21 +9,16 @@ See the `Vagrant documentation <http://docs.vagrantup.com/v2/>`_ for more detail
 Boxes
 -----
 
-There are several vagrant boxes.
+There is currently one vagrant box:
```
Comment (jml) [inline docs/gettinginvolved/infrastructure/vagrant.rst]: Actually, we're currently not supporting the Vagrant dev box either (it's broken, we've got no plans to fix it). Maybe best to get rid of both from the docs?

Leaving it up to you.

:question:

File: docs/gettinginvolved/plugins/building-driver.rst [hunk 2]
```diff
@@ -160,6 +160,4 @@ What's Next?
 
 We recommend a demo to show off your hard work!
 
-After driver development clears all tests and you've published great getting-started instructions for your users, we recommend :ref:`running through an end-to-end demo using MongoDB <tutorial-mongo>`.
-
-This demo makes a great video which you can use to share with others how they can build a Dockerized application using your storage backend.
+After driver development clears all tests and you've published great getting-started instructions for your users, we recommend a great video which you can use to share with others how they can build a Dockerized application using your storage backend.
```
Comment (jml) [inline docs/gettinginvolved/plugins/building-driver.rst]: :question: "great" is used for video & getting-started instructions. Too much?

**Comment by jml** [comment]
LGTM. A couple of suggestions.

**Comment by davidoliverSP2** [comment]
Thanks @jml 

The removal of the Vagrant Dev Box information is a bit of a stretch for this PR (which is specifically about the Tutorial). I've created FLOC-3664 to handle the removal of the dev box content.

**Pull request merged**